src/models/hooks/activityReport.js

```diff
@@ -303,7 +307,12 @@ const automaticStatusChangeOnAprovalForGoals = async (sequelize, instance, optio
         transaction: options.transaction,
       },
     );
-    await Promise.all(goals.map(async (goal) => {
+
+    // Goals updated to 'In Progress' must have at least one objective 'Completed' or 'In Progress'.
+    const goalsToUpdate = goals.filter((g) => g.objectives.some((o) => o.status === 'In Progress' || o.status === 'Completed'));
```

Comment:
Might be able to move this condition into SQL and have postgres handle it
```
          {
            model: sequelize.models.Objective,
            as: 'objectives',
            where: { status: ['In Progress', 'Completed']},
          },
```

Reply:
Yea good call. 

I had something like this initially but back tracked not understanding what they where asking for.